hyperspy/io_plugins/fei.py

```diff
@@ -580,13 +619,53 @@ def ser_reader(filename, objects=None, verbose=False, *args, **kwds):
         'mapping': mapping}
     return dictionary
 
-
-def get_mode(mode):
+def guess_units_from_mode(objects_dict, header, verbose=False):
+    # assuming that for an image stack, the UnitsLength of the "3rd" dimension
+    # is 0
+    isImageStack = (header['Dim-1_UnitsLength'][0] == 0)
+    # in case the xml file doesn't contains the "Mode" return "Unknow" unit
+    try:
+        mode = objects_dict.ObjectInfo.ExperimentalDescription.Mode
+    except AttributeError: # in case the xml chunk doesn't contain the Mode
+        return 'Unknown'
```

Comment:
Currently it would set the units to "meters" which is correct in most cases, and when it is not (diffraction) it is obvious that it is incorrect, so it is not a big deal. Return meters then?